js/ui/map.js

```diff
@@ -202,6 +203,11 @@ var Map = module.exports = function(options) {
     this.resize();
 
     if (options.classes) this.setClasses(options.classes);
+    if (options.customSourceTypes) {
+        for (var sourceTypeName in options.customSourceTypes) {
+            Source.addType(sourceTypeName, options.customSourceTypes[sourceTypeName]);
+        }
+    }
```

Comment:
👍 agreed - I removed the `customSourceTypes` option, as I think `mapboxgl.addSourceType` makes more sense.